src/main/java/org/padaiyal/utilities/aayvalar/regex/RegexUtility.java

```diff
@@ -0,0 +1,467 @@
+package org.padaiyal.utilities.aayvalar.regex;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.padaiyal.utilities.I18nUtility;
+import org.padaiyal.utilities.PropertyUtility;
+import org.padaiyal.utilities.aayvalar.regex.abstractions.Comparison;
+import org.padaiyal.utilities.aayvalar.regex.exceptions.InvalidRegexException;
+
+
+/**
+ * Utility class for manipulating regex expressions.
+ */
+public class RegexUtility {
+
+  /**
+   * Personal Identifiable Information regex expressions file name.
+   */
+  private static String regexFieldsPropertiesFileName;
+
+  /**
+   * Logger object used to log error and information.
+   */
+  private static final Logger logger = LogManager.getLogger(RegexUtility.class);
+
+  /**
+   * Word characters in a regex expression. Equivalent to \w.
+   */
+  private static final String wordCharactersRegex =
+      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_";
```

Comment:
Push all hard coded values to a property file.

Reply:
done